site/src/pages/TerminalPage/TerminalPage.test.tsx

```diff
@@ -127,7 +127,7 @@ describe("TerminalPage", () => {
     const { container } = await renderTerminal();
 
     // Then
-    await ws.connected;
```

Comment:
This is the line that seems to cause React to pause.  For example, even though the config query finishes the `useEffect` that has a dependency on it never runs again.  Swapping this to `ws.nextMessage` (to wait for the init message instead) works just fine.

Reply:
Should we add this comment somewhere in the code? I think this can be important for future maintenance.

Reply:
Good point!  Will add.